src/base/bittorrent/session.cpp

```diff
@@ -141,17 +141,66 @@ namespace
         return true;
     }
 
+#ifdef Q_OS_WIN
+    // Remove this after 4.2.5 && if at least one month has passed from v4.2.3
+    // Check the commit that introduced this function and identify all other pieces of code that
+    // need removal alongside this one.
+    void fixBrokenSavePath(QByteArray &data, lt::bdecode_node &root, ResumeDataSavingManager *resumeDataSavingManager)
+    {
+        const QString path = fromLTString(root.dict_find_string_value("save_path"));
+        const int index = path.indexOf(QLatin1String("//"));
+        if (index < 1)
+            return;
+        const QString goodPath = path.mid(index).replace('/', '\\');
+        lt::entry entry(root);
+        entry["save_path"] = goodPath.toStdString();
+
+        const auto rawView = root.dict_find_string_value("info-hash");
+        const QByteArray rawHashView = QByteArray::fromRawData(rawView.data(), rawView.length());
+        const QString hexHash = QString::fromLatin1(rawHashView.toHex());
+
+        data.clear();
+        lt::bencode(std::back_inserter(data), entry);
+
+        const QString filename = QString("%1.fastresume").arg(hexHash);
```

Comment:
I would write it to file here without any saving manager. This may slow down the app loading somewhat, but it will actually happen only once.

Reply:
Done.
(Although I think it is a good idea to let the specialized systems handle a task)